styleguide/source/assets/js/webforms.js

```diff
@@ -1,110 +1,45 @@
-    var $iconElement = $('.ama__form-steps__icon');
-    var errorSections = [];
-
-    $inputs.each(function(i, input) {
-      $closestSection = $(this).closest('section').attr('data-drupal-selector').toString();
-      if ($(this).prop('required') && $(this).hasClass('error')) {
-        errorSections.push($closestSection);
-      }
-    });
-
-    $sections.each(function(i, section) {
-      if ($.inArray($(this).attr('data-drupal-selector').trim().toString(), errorSections) !== -1) {
-        $(this).find($iconElement).removeClass('edit error completed').addClass('error');
-      }
-      else {
-        $(this).find($iconElement).removeClass('edit error completed').addClass('completed');
-      }
-    });
-    return this;
-  };
-
-  function validateEmail(email) {
-    var emailReg = /^([\w-\.]+@([\w-]+\.)+[\w-]{2,4})?$/;
-    return emailReg.test(email);
-  }
-
-  function fieldIsRequired(input) {
-    input.addClass('error');
-    input.next().remove('.form-item--error-message');
-    input.after('<div class="form-item--error-message">Field is required.</div>');
-  }
-
-  function checkField(input) {
-    if (input.prop('required') && (input.val().length === 0 || input.val() === "")) {
-      fieldIsRequired(input);
-    }
-    else {
-      if (input.attr('type') === 'email' && !validateEmail(input.val())) {
-        fieldIsRequired(input);
-      }
-      else {
-        input.removeClass('error').next().remove('.form-item--error-message');
-      }
-    }
-  }
-
-  var initialLoad = true;
-
-  Drupal.behaviors.webForm = {
+  Drupal.behaviors.formValidate = {
     attach: function (context, settings) {
-      $(document).ready(function(e) {
-        $contactForm = $('.webform-submission-form');
-        $inputs = $('.webform-submission-form section *').filter(':input');
-        $iconElement = $('.ama__form-steps__icon');
-        $submitBuutton = $('.webform-button--submit');
-
-        if (initialLoad === false) {
-          verifyFields($contactForm);
-        }
-
-        $inputs.on('focus change keypress selectmenuchange', function () {
-          var iconClass = 'edit';
-          $closestSection = $(this).closest('section');
-          $closestSectionInputs = $closestSection.find(':input');
-          $closestSection.find($iconElement).removeClass('edit error completed').addClass(iconClass);
-          $closestSectionInputs.each(function () {
-            checkField($(this));
-          });
-        });
-
-        $inputs.on('focus blur', function () {
-          var iconClass = 'edit';
-          $closestSection = $(this).closest('section');
-          $closestSectionInputs = $closestSection.find(':input');
-          $allFieldsReady = true;
-
-          $closestSectionInputs.each(function () {
-            if ($(this).prop('required') && $(this).val().length === 0) {
-              $allFieldsReady = false;
-              iconClass = 'error';
+      (function ($) {
```

Comment:
jQuery is already scoped. If we are using this for $(document).ready(), that isn't needed since Drupal only calls the behavior methods when the DOM is fully loaded. See https://www.lullabot.com/articles/understanding-javascript-behaviors-in-drupal.